scripts/build_rust_artifacts.py

```diff
@@ -7,53 +7,88 @@
 import shlex
 
 
-def cmd(command):
+def run(command):
     assert isinstance(command, str)
-    # This allows us to avoid manually splitting strings into arguments
-    command = shlex.split(command)
-    print(f"Running: {' '.join(command)}")
-    subprocess.check_call(command)
+    print(f"Running: {command}")
+    exit_code = subprocess.check_call(command, shell=True, text=True)
+    assert exit_code == 0
 
 
-def main():
-    parser = argparse.ArgumentParser(description="Builds rust artifacts for CI")
-    parser.add_argument("out_dir", help="Output directory for artifacts")
-    parser.add_argument(
-        "cargo_profile", help="Cargo profile to use for compiling the crates"
-    )
-    parser.add_argument("crates", nargs="+", help="List of crate names to be processed")
+def run_with_stdout(command):
+    assert isinstance(command, str)
+    print(f"Running: {command}")
+    cmd_output = subprocess.check_output(command, shell=True, text=True)
+    return cmd_output
 
-    args = parser.parse_args()
 
-    targets = ["aarch64", "x86_64"]
+def find_cargo_deb_crates():
+    jq_query = (
```

Comment:
The alternative in Python would be (something along these lines):
```
    metadata = json.loads(cmd_output)
    workspace_members = set(metadata['workspace_members'])
    crates = [
        package['name']
        for package in metadata['packages']
        if package['id'] in workspace_members and 'deb' in package.get('metadata', {})
    ]
```

I personally find it more understandable. But I'll leave this to you. 

Is `jq` always available btw? Because with pure python implementation you have the advantage of "peace of mind".

Reply:
we ensure that jq is always available via the nix flake. But yes, I think I prefer the python approach more. Also, I know python more than I know jq. I'll open a new PR to port it.